pkg/storaged/drives-panel.jsx

```diff
@@ -0,0 +1,123 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+import React from "react";
+
+import { OverviewSidePanel, OverviewSidePanelRow } from "./overview.jsx";
+import { fmt_size, drive_name, decode_filename } from "./utils.js";
+
+const _ = cockpit.gettext;
+const C_ = cockpit.gettext;
+
+export class DrivesPanel extends React.Component {
+    constructor () {
+        super();
+        this.on_io_samples = () => { this.setState({}); }
+    }
+
+    componentDidMount() {
+        this.props.client.blockdev_io.addEventListener("changed", this.on_io_samples);
+    }
+
+    componentWillUnmount() {
+        this.props.client.blockdev_io.removeEventListener("changed", this.on_io_samples);
+    }
+
+    render() {
+        var self = this;
```

Comment:
This pattern should really be avoided in ES6 with arrow functions. You don't need it at all here, the only usage of `self` is in this very render() function.

Reply:
Yep.  I still have to get used to using arrows for non-anonymous functions...  Fixed.